src/scheduler/node_info.c

```diff
@@ -6265,6 +6321,42 @@ check_node_eligibility_chunk(th_data_nd_eligible *data)
 	data->err = misc_err;
 }
 
+/**
+ * @brief	 Allocates th_data_nd_eligible for multi-threading of check_node_array_eligibility
+ *
+ * @param[in]	pl	-	the placement object
+ * @param[in]	resresv	-	resresv to check to place on nodes
+ * @param[in]	exclerr_buf	-	buffer for not exclusive error message
+ * @param[in]	ninfo_arr	-	array to check
+ * @param[in]	sidx	-	the start index in ninfo_arr for the thread
+ * @param[in]	eidx	-	the end index in ninfo_arr for the thread
+ *
+ * @return th_data_nd_eligible *
+ * @retval a newly allocated th_data_nd_eligible object
+ * @retval NULL for malloc error
+ */
+static th_data_nd_eligible *
+alloc_tdata_nd_eligible(place *pl, resource_resv *resresv, char *exclerr_buf, node_info **ninfo_arr,
+		int sidx, int eidx)
+{
+	th_data_nd_eligible *tdata = NULL;
+
+	tdata = malloc(sizeof(th_data_nd_eligible));
+	if (tdata == NULL)  {
+		log_err(errno, __func__, MEM_ERR_MSG);
+		return NULL;
+	}
+	tdata->err = NULL;
+	tdata->pl = pl;
+	tdata->resresv = resresv;
+	tdata->exclerr_buf = exclerr_buf;
```

Comment:
I thought I made a comment to not do pass this in.  Just translate it if you need it.

Reply:
I must have missed it, will make the change thanks.